{
    "expand": "operations,versionedRepresentations,editmeta,changelog,renderedFields",
    "fields": {
        "aggregateprogress": {
            "progress": 0,
            "total": 0
        },
        "aggregatetimeestimate": null,
        "aggregatetimeoriginalestimate": null,
        "aggregatetimespent": null,
        "assignee": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=stevel%40apache.org&avatarId=16513",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=stevel%40apache.org&avatarId=16513",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=stevel%40apache.org&avatarId=16513",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=stevel%40apache.org&avatarId=16513"
            },
            "displayName": "Steve Loughran",
            "key": "stevel@apache.org",
            "name": "stevel@apache.org",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=stevel%40apache.org",
            "timeZone": "Europe/London"
        },
        "components": [{
            "description": "S3A filesystem client and other S3 connectivity issues",
            "id": "12311814",
            "name": "fs/s3",
            "self": "https://issues.apache.org/jira/rest/api/2/component/12311814"
        }],
        "created": "2013-01-29T23:25:41.000+0000",
        "creator": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=stevel%40apache.org&avatarId=16513",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=stevel%40apache.org&avatarId=16513",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=stevel%40apache.org&avatarId=16513",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=stevel%40apache.org&avatarId=16513"
            },
            "displayName": "Steve Loughran",
            "key": "stevel@apache.org",
            "name": "stevel@apache.org",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=stevel%40apache.org",
            "timeZone": "Europe/London"
        },
        "customfield_10010": null,
        "customfield_12310191": null,
        "customfield_12310192": null,
        "customfield_12310220": "2013-01-30T19:47:39.707+0000",
        "customfield_12310222": "3_*:*_1_*:*_4059523_*|*_10002_*:*_2_*:*_549052599_*|*_1_*:*_3_*:*_4162147504_*|*_5_*:*_1_*:*_0",
        "customfield_12310230": null,
        "customfield_12310250": null,
        "customfield_12310290": null,
        "customfield_12310291": null,
        "customfield_12310300": null,
        "customfield_12310310": "2.0",
        "customfield_12310320": null,
        "customfield_12310420": "310308",
        "customfield_12310920": "310653",
        "customfield_12310921": null,
        "customfield_12311020": null,
        "customfield_12311024": null,
        "customfield_12311120": null,
        "customfield_12311820": "0|i1hjxb:",
        "customfield_12312022": null,
        "customfield_12312023": null,
        "customfield_12312024": null,
        "customfield_12312026": null,
        "customfield_12312220": null,
        "customfield_12312320": null,
        "customfield_12312321": null,
        "customfield_12312322": null,
        "customfield_12312323": null,
        "customfield_12312324": null,
        "customfield_12312325": null,
        "customfield_12312326": null,
        "customfield_12312327": null,
        "customfield_12312328": null,
        "customfield_12312329": null,
        "customfield_12312330": null,
        "customfield_12312331": null,
        "customfield_12312332": null,
        "customfield_12312333": null,
        "customfield_12312334": null,
        "customfield_12312335": null,
        "customfield_12312336": null,
        "customfield_12312337": null,
        "customfield_12312338": null,
        "customfield_12312339": null,
        "customfield_12312340": null,
        "customfield_12312341": null,
        "customfield_12312520": null,
        "customfield_12312521": "Fri Nov 13 18:53:23 UTC 2015",
        "customfield_12312720": null,
        "customfield_12312823": null,
        "customfield_12312920": null,
        "customfield_12312921": null,
        "customfield_12312923": null,
        "customfield_12313422": "false",
        "customfield_12313520": null,
        "description": "The S3N filesystem {{rename()}} doesn't make sure that the destination directory is not a child or other descendant of the source directory. The files are copied to the new destination, then the source directory is recursively deleted, so losing data.",
        "duedate": null,
        "environment": "Testing against S3 bucket stored on US West (Read after Write consistency; eventual for read-after-delete or write-after-write)",
        "fixVersions": [{
            "archived": false,
            "description": "3.0.0-alpha1 release",
            "id": "12335733",
            "name": "3.0.0-alpha1",
            "releaseDate": "2016-09-03",
            "released": true,
            "self": "https://issues.apache.org/jira/rest/api/2/version/12335733"
        }],
        "issuelinks": [
            {
                "id": "12364841",
                "inwardIssue": {
                    "fields": {
                        "issuetype": {
                            "avatarId": 21146,
                            "description": "The sub-task of the issue",
                            "iconUrl": "https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21146&avatarType=issuetype",
                            "id": "7",
                            "name": "Sub-task",
                            "self": "https://issues.apache.org/jira/rest/api/2/issuetype/7",
                            "subtask": true
                        },
                        "priority": {
                            "iconUrl": "https://issues.apache.org/jira/images/icons/priorities/major.svg",
                            "id": "3",
                            "name": "Major",
                            "self": "https://issues.apache.org/jira/rest/api/2/priority/3"
                        },
                        "status": {
                            "description": "The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.",
                            "iconUrl": "https://issues.apache.org/jira/images/icons/statuses/closed.png",
                            "id": "6",
                            "name": "Closed",
                            "self": "https://issues.apache.org/jira/rest/api/2/status/6",
                            "statusCategory": {
                                "colorName": "green",
                                "id": 3,
                                "key": "done",
                                "name": "Done",
                                "self": "https://issues.apache.org/jira/rest/api/2/statuscategory/3"
                            }
                        },
                        "summary": "Add stricter tests to FileSystemContractTestBase"
                    },
                    "id": "12629587",
                    "key": "HADOOP-9258",
                    "self": "https://issues.apache.org/jira/rest/api/2/issue/12629587"
                },
                "self": "https://issues.apache.org/jira/rest/api/2/issueLink/12364841",
                "type": {
                    "id": "12310060",
                    "inward": "Is contained by",
                    "name": "Container",
                    "outward": "contains",
                    "self": "https://issues.apache.org/jira/rest/api/2/issueLinkType/12310060"
                }
            },
            {
                "id": "12477949",
                "inwardIssue": {
                    "fields": {
                        "issuetype": {
                            "avatarId": 21148,
                            "description": "A task that needs to be done.",
                            "iconUrl": "https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21148&avatarType=issuetype",
                            "id": "3",
                            "name": "Task",
                            "self": "https://issues.apache.org/jira/rest/api/2/issuetype/3",
                            "subtask": false
                        },
                        "priority": {
                            "iconUrl": "https://issues.apache.org/jira/images/icons/priorities/minor.svg",
                            "id": "4",
                            "name": "Minor",
                            "self": "https://issues.apache.org/jira/rest/api/2/priority/4"
                        },
                        "status": {
                            "description": "A resolution has been taken, and it is awaiting verification by reporter. From here issues are either reopened, or are closed.",
                            "iconUrl": "https://issues.apache.org/jira/images/icons/statuses/resolved.png",
                            "id": "5",
                            "name": "Resolved",
                            "self": "https://issues.apache.org/jira/rest/api/2/status/5",
                            "statusCategory": {
                                "colorName": "green",
                                "id": 3,
                                "key": "done",
                                "name": "Done",
                                "self": "https://issues.apache.org/jira/rest/api/2/statuscategory/3"
                            }
                        },
                        "summary": "backport HADOOP-9258 to branch-2"
                    },
                    "id": "12998279",
                    "key": "HADOOP-13518",
                    "self": "https://issues.apache.org/jira/rest/api/2/issue/12998279"
                },
                "self": "https://issues.apache.org/jira/rest/api/2/issueLink/12477949",
                "type": {
                    "id": "10030",
                    "inward": "is related to",
                    "name": "Reference",
                    "outward": "relates to",
                    "self": "https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"
                }
            }
        ],
        "issuetype": {
            "avatarId": 21133,
            "description": "A problem which impairs or prevents the functions of the product.",
            "iconUrl": "https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype",
            "id": "1",
            "name": "Bug",
            "self": "https://issues.apache.org/jira/rest/api/2/issuetype/1",
            "subtask": false
        },
        "labels": [],
        "lastViewed": null,
        "priority": {
            "iconUrl": "https://issues.apache.org/jira/images/icons/priorities/major.svg",
            "id": "3",
            "name": "Major",
            "self": "https://issues.apache.org/jira/rest/api/2/priority/3"
        },
        "progress": {
            "progress": 0,
            "total": 0
        },
        "project": {
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=12310240&avatarId=10095",
                "24x24": "https://issues.apache.org/jira/secure/projectavatar?size=small&pid=12310240&avatarId=10095",
                "32x32": "https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=12310240&avatarId=10095",
                "48x48": "https://issues.apache.org/jira/secure/projectavatar?pid=12310240&avatarId=10095"
            },
            "id": "12310240",
            "key": "HADOOP",
            "name": "Hadoop Common",
            "projectCategory": {
                "description": "Scalable Distributed Computing",
                "id": "10292",
                "name": "Hadoop",
                "self": "https://issues.apache.org/jira/rest/api/2/projectCategory/10292"
            },
            "self": "https://issues.apache.org/jira/rest/api/2/project/12310240"
        },
        "reporter": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=stevel%40apache.org&avatarId=16513",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=stevel%40apache.org&avatarId=16513",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=stevel%40apache.org&avatarId=16513",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=stevel%40apache.org&avatarId=16513"
            },
            "displayName": "Steve Loughran",
            "key": "stevel@apache.org",
            "name": "stevel@apache.org",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=stevel%40apache.org",
            "timeZone": "Europe/London"
        },
        "resolution": {
            "description": "A fix for this issue is checked into the tree and tested.",
            "id": "1",
            "name": "Fixed",
            "self": "https://issues.apache.org/jira/rest/api/2/resolution/1"
        },
        "resolutiondate": "2013-03-25T13:13:21.000+0000",
        "status": {
            "description": "A resolution has been taken, and it is awaiting verification by reporter. From here issues are either reopened, or are closed.",
            "iconUrl": "https://issues.apache.org/jira/images/icons/statuses/resolved.png",
            "id": "5",
            "name": "Resolved",
            "self": "https://issues.apache.org/jira/rest/api/2/status/5",
            "statusCategory": {
                "colorName": "green",
                "id": 3,
                "key": "done",
                "name": "Done",
                "self": "https://issues.apache.org/jira/rest/api/2/statuscategory/3"
            }
        },
        "subtasks": [],
        "summary": "S3n filesystem can move a directory under itself -and so lose data",
        "timeestimate": null,
        "timeoriginalestimate": null,
        "timespent": null,
        "updated": "2016-08-18T18:35:38.000+0000",
        "versions": [
            {
                "archived": false,
                "description": "maintenance release on branch-1.1",
                "id": "12321658",
                "name": "1.1.1",
                "releaseDate": "2012-11-27",
                "released": true,
                "self": "https://issues.apache.org/jira/rest/api/2/version/12321658"
            },
            {
                "archived": false,
                "description": "2.0.2-alpha release",
                "id": "12322473",
                "name": "2.0.2-alpha",
                "releaseDate": "2012-10-09",
                "released": true,
                "self": "https://issues.apache.org/jira/rest/api/2/version/12322473"
            }
        ],
        "votes": {
            "hasVoted": false,
            "self": "https://issues.apache.org/jira/rest/api/2/issue/HADOOP-9261/votes",
            "votes": 0
        },
        "watches": {
            "isWatching": false,
            "self": "https://issues.apache.org/jira/rest/api/2/issue/HADOOP-9261/watchers",
            "watchCount": 5
        },
        "workratio": -1
    },
    "id": "12629812",
    "key": "HADOOP-9261",
    "self": "https://issues.apache.org/jira/rest/api/2/issue/12629812"
}